### PR TITLE
truncate metadata entries to 512 characters

### PR DESCRIPTION
**Pull request by coreykosak**

### CODE REVIEW

File: src/write_gcm.c [hunk 1]
```diff
@@ -1423,7 +1423,14 @@ static int wg_typed_value_create_from_meta_data_inline(wg_typed_value_t *result,
     case MD_TYPE_STRING: {
       result->field_name_static = "stringValue";
       result->value_type = wg_typed_value_string;
-      return meta_data_get_string(md, key, &result->value_text);
+      if (meta_data_get_string(md, key, &result->value_text) != 0) {
+        return -1;
+      }
+      // Truncate all metadata entries to 512 characters.
+      if (sstrlen(result->value_text) > 512) {
+        result->value_text[512] = 0;
```
Comment (igorpeshansky) [inline src/write_gcm.c]: Let's make it a character ('\0')?

File: src/write_gcm.c [hunk 2]
```diff
@@ -1423,7 +1423,14 @@ static int wg_typed_value_create_from_meta_data_inline(wg_typed_value_t *result,
     case MD_TYPE_STRING: {
       result->field_name_static = "stringValue";
       result->value_type = wg_typed_value_string;
-      return meta_data_get_string(md, key, &result->value_text);
+      if (meta_data_get_string(md, key, &result->value_text) != 0) {
+        return -1;
+      }
+      // Truncate all metadata entries to 512 characters.
+      if (sstrlen(result->value_text) > 512) {
```
Comment (igorpeshansky) [inline src/write_gcm.c]: What is this sstrlen you speak of?

**Comment by igorpeshansky** [comment]
Let's wait for the results of the internal discussion before merging...

File: src/write_gcm.c (hunk 1 above)
Comment (coreykosak) [inline src/write_gcm.c]: I don't particularly like '\0' but if you insist. Done.

**Comment by coreykosak** [comment]
OK but sounds like Kilby has already updated the Windows side.

**Comment by igorpeshansky** [comment]
Ok, let's merge it.

**Pull request merged**